rvgo/fast/state.go

```diff
@@ -3,6 +3,11 @@ package fast
 import (
 	"encoding/binary"
 	"fmt"
+	"io"
+
+	"github.com/ethereum-optimism/optimism/op-service/jsonutil"
+
+	"github.com/ethereum-optimism/optimism/cannon/serialize"
```

Comment:
I hadn't really thought of the binary serialisation as a reusable library though I can see the value (the BinaryWriter struct got extracted out from PR feedback so I didn't really think a lot about where it lived).

I wonder if we should move the reused parts out of cannon into op-service? It's a bit weird to have asterisc depend on cannon and having it in op-service makes it a bit clearer that it's intended to be reused so we're less likely to randomly change it on you based on what cannon needs (which may then cause backwards compatibility problems for asterisc).

Reply:
I agree that it would be much better if we could isolate this serialization logic into op-service like you've mentioned!(although we do have many dependency of cannon in various parts of asterisc 😅 ) 

Then, I'll make that change in the monorepo as a separate PR (so that the other op-challenger diff doesn't get mixed). Thanks!

Reply:
Over time would be great to remove the dependencies on cannon - both from asterisc and op-challenger.  I think we're fairly close for challenger, it's mostly just the exit code constants and I'm just not really sure where to put them. :)

Reply:
I created a PR here: https://github.com/ethereum-optimism/optimism/pull/12265. Would love to have your review on it :)